core/src/test/java/org/infinispan/lock/LockContainerTest.java

```diff
@@ -32,29 +30,28 @@
  */
 @Test(groups = "unit", testName = "lock.LockContainerTest")
 public class LockContainerTest extends AbstractInfinispanTest {
-   private final ExecutorService executor = new WithinThreadExecutor();
 
    public void testSingleLockWithPerEntry() throws InterruptedException {
       PerKeyLockContainer lockContainer = new PerKeyLockContainer();
-      TestingUtil.inject(lockContainer, executor, AbstractCacheTest.TIME_SERVICE);
+      TestingUtil.inject(lockContainer, AbstractCacheTest.TIME_SERVICE);
```

Comment:
I'm a bit scared that none of these tests need the executor, it suggests that maybe they don't cover all the code paths ;)

Reply:
I agree. As I mentioned above the executor is only used when it has to wait for the lock.